src/state/slices/marketDataSlice/marketDataSlice.tsx

```diff
@@ -31,30 +50,50 @@ const initialPriceHistory: PriceHistoryByTimeframe = {
 }
 
 const initialState: MarketDataState = {
-  byId: {},
-  ids: [],
-  priceHistory: initialPriceHistory,
-  loading: false,
+  crypto: {
+    byId: {},
+    ids: [],
+    priceHistory: initialPriceHistory,
+  },
+  fiat: {
+    byId: {},
+    ids: [],
+    priceHistory: initialPriceHistory,
+  },
```

Comment:
this is the crux of the PR - we're adding fiat price history to the market data state, and namespacing under `crypto` and `fiat` respectively. everything else is a result of this.